addons/project/wizard/project_share_wizard.py

```diff
@@ -0,0 +1,81 @@
+            project = self.env[result['res_model']].browse(result['res_id'])
+            result.update(
+                allow_project_sharing=project.allow_project_sharing,
+                access_ids=[
+                    Command.create(
+                        {'user_id': access.user_id, 'access_mode': access.access_mode}
+                    ) for access in project.project_sharing_access_ids
+                ],
+            )
+        return result
+
+    partner_ids = fields.Many2many(compute="_compute_partner_ids", readonly=False, store=True, required=False)
+    access_ids = fields.One2many('project.share.access.wizard', 'project_share_id')
+    allow_project_sharing = fields.Boolean(default=False)
+
+    @api.depends('access_ids.user_id')
+    def _compute_partner_ids(self):
+        for project_share_wizard in self:
+            project_share_wizard.partner_ids |= project_share_wizard.access_ids.user_id.partner_id
+
+    def action_send_mail_and_confirm_access(self):
+        if not self.partner_ids:
+            raise UserError(_('Impossible to send email, no recipients are mentionned.'))
+        self.action_confirm_access()
+        return self.action_send_mail()
+
+    def action_confirm_access(self):
+        project = self.env[self.res_model].browse(self.res_id)
+        if not project or not project.exists() or not project.allow_project_sharing:
+            raise ValidationError(_('Impossible to confirm the project sharing access, the project is not found or project sharing is disabled for this project.'))
+        project_sharing_access_dict = {access.user_id.id: access for access in project.project_sharing_access_ids}
+        project_sharing_access_vals_list = [
+            Command.unlink(access.id)
+            for user_id, access in project_sharing_access_dict.items()
+            if user_id not in self.access_ids.user_id.ids
+        ]
+        for share_access in self.access_ids:
+            project_sharing_access = project_sharing_access_dict.get(share_access.user_id.id)
+            if not project_sharing_access:
+                project_sharing_access_vals_list.append(
+                    Command.create(
+                        {'user_id': share_access.user_id.id, 'access_mode': share_access.access_mode}
+                    )
+                )
+            elif project_sharing_access.access_mode != share_access.access_mode:
+                project_sharing_access_vals_list.append(Command.update(project_sharing_access.id, {'access_mode': share_access.access_mode}))
+        project.write({'project_sharing_access_ids': project_sharing_access_vals_list})
+        # FIXME: See if we directly add the portal user with access to project sharing as followers to the shared project
+        project.message_subscribe(self.access_ids.user_id.partner_id.ids)
+        return {'type': 'ir.actions.act_window_close'}
+
+
+class ProjectShareAccessWizard(models.TransientModel):
+    _name = 'project.share.access.wizard'
```

Comment:
```suggestion
    _name = 'project.share.wizard.user'
```
or 
```suggestion
    _name = 'project.share.wizard.line'
```
+ fix description